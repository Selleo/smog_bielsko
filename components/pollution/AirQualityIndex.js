import React, { Component, PropTypes } from 'react';
import { View, StyleSheet, Text } from 'react-native';

export default class AirQualityIndex extends Component {
  render() {
    var index = this.getAirConditionIndex(this.props.index);
    return (
      <View style={styles.container} backgroundColor={colors[index]}>
        <Text style={styles.text}>Jakość powietrza: {names[index]}</Text>
      </View>
    )
  }

  getAirConditionIndex(aqi) {
    if (aqi < 50) { return 0 }
    if (aqi < 100) { return 1 }
    if (aqi < 150) { return 2 }
    if (aqi < 200) { return 3 }
    if (aqi < 300) { return 4 }
    return 5
  }

  static propTypes = {
    index: PropTypes.number.isRequired
  };
}

const names = [
  'Dobra',
  'Średnia',
  'Kiepska',
  'Zła',
  'Bardzo zła',
  'Zagrożenie życia'
];

const colors = [
  '#2DB338',
  '#86B32D',
  '#B3AA2D',
  '#B5753E',
  '#B34F2D',
  '#7D6363'
];

const styles = StyleSheet.create({
  text: {
    fontSize: 16,
    paddingHorizontal: 10,
    paddingVertical: 6,
    color: '#111'
  }
});
