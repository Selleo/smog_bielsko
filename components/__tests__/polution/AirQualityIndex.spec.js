import 'react-native';
import React                          from 'react';

import { shallow }                    from 'enzyme';
import chai, { expect }               from 'chai';
import dirtyChai                      from 'dirty-chai';

import AirQualityIndex                 from '../../pollution/AirQualityIndex';

const originalExpect = global.expect;

chai.use(dirtyChai);

let wrapper, props;

describe('AirQualityIndex', () => {
  beforeEach(() => {
    props = {
      index: 1
    };

    wrapper = shallow(<AirQualityIndex {...props}/>);
  });

  it('renders correctly', () => {
    originalExpect(wrapper).toMatchSnapshot();
    expect(wrapper).to.exist();
  });

  it('render staton name on html', () => {
    expect(wrapper.children().children().nodes.join('')).to.equal('Jakość powietrza: Dobra')
  });

  describe('#getAirConditionIndex', () => {
    it('return 0', () => {
      expect(wrapper.instance().getAirConditionIndex(30)).to.equal(0)
    });

    it('return 1', () => {
      expect(wrapper.instance().getAirConditionIndex(80)).to.equal(1)
    });

    it('return 2', () => {
      expect(wrapper.instance().getAirConditionIndex(110)).to.equal(2)
    });

    it('return 3', () => {
      expect(wrapper.instance().getAirConditionIndex(160)).to.equal(3)
    });

    it('return 4', () => {
      expect(wrapper.instance().getAirConditionIndex(230)).to.equal(4)
    });

    it('return 5', () => {
      expect(wrapper.instance().getAirConditionIndex(350)).to.equal(5)
    });
  });
});
