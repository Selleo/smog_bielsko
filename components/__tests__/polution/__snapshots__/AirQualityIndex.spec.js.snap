exports[`AirQualityIndex renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Array [
            Object {
              "alignItems": "stretch",
              "backgroundColor": "white",
              "borderRadius": 3,
            },
            Object {
              "paddingTop": 5,
            },
            Object {
              "backgroundColor": "white",
              "marginTop": 15,
            },
          ]
    }>
    <ElevatedView
        elevation={4}
        style={
            Array [
                Object {
                  "alignItems": "stretch",
                  "backgroundColor": "white",
                  "justifyContent": "space-between",
                  "margin": 20,
                  "marginBottom": 15,
                  "marginTop": 0,
                },
                Object {
                  "alignItems": "center",
                },
                Object {
                  "padding": 5,
                },
              ]
        }>
        <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                Object {
                    "color": "#111",
                    "fontSize": 14,
                    "paddingHorizontal": 10,
                    "paddingTop": 3,
                  }
            }>
            Jakość powietrza:
        </Text>
        <View
            style={
                Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "flexWrap": "nowrap",
                    "justifyContent": "center",
                  }
            }>
            <View
                backgroundColor="#2DB338"
                style={
                    Object {
                        "alignItems": "center",
                        "borderRadius": 30,
                        "height": 15,
                        "justifyContent": "center",
                        "overflow": "hidden",
                        "width": 15,
                      }
                }>
                <View
                    style={
                        Object {
                            "alignItems": "center",
                            "justifyContent": "center",
                          }
                    } />
            </View>
            <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                    Object {
                        "color": "#111",
                        "fontSize": 24,
                        "paddingBottom": 3,
                        "paddingHorizontal": 10,
                      }
                }>
                Dobra
            </Text>
            <View
                backgroundColor="#2DB338"
                style={
                    Object {
                        "alignItems": "center",
                        "borderRadius": 30,
                        "height": 15,
                        "justifyContent": "center",
                        "overflow": "hidden",
                        "width": 15,
                      }
                }>
                <View
                    style={
                        Object {
                            "alignItems": "center",
                            "justifyContent": "center",
                          }
                    } />
            </View>
        </View>
    </ElevatedView>
</View>,
  "nodes": Array [
    <View
      style={
            Array [
                  Object {
                    "alignItems": "stretch",
                    "backgroundColor": "white",
                    "borderRadius": 3,
                  },
                  Object {
                    "paddingTop": 5,
                  },
                  Object {
                    "backgroundColor": "white",
                    "marginTop": 15,
                  },
                ]
      }>
      <ElevatedView
            elevation={4}
            style={
                  Array [
                        Object {
                          "alignItems": "stretch",
                          "backgroundColor": "white",
                          "justifyContent": "space-between",
                          "margin": 20,
                          "marginBottom": 15,
                          "marginTop": 0,
                        },
                        Object {
                          "alignItems": "center",
                        },
                        Object {
                          "padding": 5,
                        },
                      ]
            }>
            <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                        Object {
                              "color": "#111",
                              "fontSize": 14,
                              "paddingHorizontal": 10,
                              "paddingTop": 3,
                            }
                  }>
                  Jakość powietrza:
            </Text>
            <View
                  style={
                        Object {
                              "alignItems": "center",
                              "flex": 1,
                              "flexDirection": "row",
                              "flexWrap": "nowrap",
                              "justifyContent": "center",
                            }
                  }>
                  <View
                        backgroundColor="#2DB338"
                        style={
                              Object {
                                    "alignItems": "center",
                                    "borderRadius": 30,
                                    "height": 15,
                                    "justifyContent": "center",
                                    "overflow": "hidden",
                                    "width": 15,
                                  }
                        }>
                        <View
                              style={
                                    Object {
                                          "alignItems": "center",
                                          "justifyContent": "center",
                                        }
                              } />
                  </View>
                  <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                              Object {
                                    "color": "#111",
                                    "fontSize": 24,
                                    "paddingBottom": 3,
                                    "paddingHorizontal": 10,
                                  }
                        }>
                        Dobra
                  </Text>
                  <View
                        backgroundColor="#2DB338"
                        style={
                              Object {
                                    "alignItems": "center",
                                    "borderRadius": 30,
                                    "height": 15,
                                    "justifyContent": "center",
                                    "overflow": "hidden",
                                    "width": 15,
                                  }
                        }>
                        <View
                              style={
                                    Object {
                                          "alignItems": "center",
                                          "justifyContent": "center",
                                        }
                              } />
                  </View>
            </View>
      </ElevatedView>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <AirQualityIndex
        index={1} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": AirQualityIndex {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onLayout": [Function],
        "props": Object {
          "index": 1,
        },
        "refs": Object {},
        "state": Object {
          "height": 1334,
          "width": 750,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Array [
                              Object {
                                "alignItems": "stretch",
                                "backgroundColor": "white",
                                "borderRadius": 3,
                              },
                              Object {
                                "paddingTop": 5,
                              },
                              Object {
                                "backgroundColor": "white",
                                "marginTop": 15,
                              },
                            ]
          }>
          <ElevatedView
                    elevation={4}
                    style={
                              Array [
                                        Object {
                                          "alignItems": "stretch",
                                          "backgroundColor": "white",
                                          "justifyContent": "space-between",
                                          "margin": 20,
                                          "marginBottom": 15,
                                          "marginTop": 0,
                                        },
                                        Object {
                                          "alignItems": "center",
                                        },
                                        Object {
                                          "padding": 5,
                                        },
                                      ]
                    }>
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#111",
                                                  "fontSize": 14,
                                                  "paddingHorizontal": 10,
                                                  "paddingTop": 3,
                                                }
                              }>
                              Jakość powietrza:
                    </Text>
                    <View
                              style={
                                        Object {
                                                  "alignItems": "center",
                                                  "flex": 1,
                                                  "flexDirection": "row",
                                                  "flexWrap": "nowrap",
                                                  "justifyContent": "center",
                                                }
                              }>
                              <View
                                        backgroundColor="#2DB338"
                                        style={
                                                  Object {
                                                            "alignItems": "center",
                                                            "borderRadius": 30,
                                                            "height": 15,
                                                            "justifyContent": "center",
                                                            "overflow": "hidden",
                                                            "width": 15,
                                                          }
                                        }>
                                        <View
                                                  style={
                                                            Object {
                                                                      "alignItems": "center",
                                                                      "justifyContent": "center",
                                                                    }
                                                  } />
                              </View>
                              <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        ellipsizeMode="tail"
                                        style={
                                                  Object {
                                                            "color": "#111",
                                                            "fontSize": 24,
                                                            "paddingBottom": 3,
                                                            "paddingHorizontal": 10,
                                                          }
                                        }>
                                        Dobra
                              </Text>
                              <View
                                        backgroundColor="#2DB338"
                                        style={
                                                  Object {
                                                            "alignItems": "center",
                                                            "borderRadius": 30,
                                                            "height": 15,
                                                            "justifyContent": "center",
                                                            "overflow": "hidden",
                                                            "width": 15,
                                                          }
                                        }>
                                        <View
                                                  style={
                                                            Object {
                                                                      "alignItems": "center",
                                                                      "justifyContent": "center",
                                                                    }
                                                  } />
                              </View>
                    </View>
          </ElevatedView>
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Array [
                              Object {
                                "alignItems": "stretch",
                                "backgroundColor": "white",
                                "borderRadius": 3,
                              },
                              Object {
                                "paddingTop": 5,
                              },
                              Object {
                                "backgroundColor": "white",
                                "marginTop": 15,
                              },
                            ]
          }>
          <ElevatedView
                    elevation={4}
                    style={
                              Array [
                                        Object {
                                          "alignItems": "stretch",
                                          "backgroundColor": "white",
                                          "justifyContent": "space-between",
                                          "margin": 20,
                                          "marginBottom": 15,
                                          "marginTop": 0,
                                        },
                                        Object {
                                          "alignItems": "center",
                                        },
                                        Object {
                                          "padding": 5,
                                        },
                                      ]
                    }>
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#111",
                                                  "fontSize": 14,
                                                  "paddingHorizontal": 10,
                                                  "paddingTop": 3,
                                                }
                              }>
                              Jakość powietrza:
                    </Text>
                    <View
                              style={
                                        Object {
                                                  "alignItems": "center",
                                                  "flex": 1,
                                                  "flexDirection": "row",
                                                  "flexWrap": "nowrap",
                                                  "justifyContent": "center",
                                                }
                              }>
                              <View
                                        backgroundColor="#2DB338"
                                        style={
                                                  Object {
                                                            "alignItems": "center",
                                                            "borderRadius": 30,
                                                            "height": 15,
                                                            "justifyContent": "center",
                                                            "overflow": "hidden",
                                                            "width": 15,
                                                          }
                                        }>
                                        <View
                                                  style={
                                                            Object {
                                                                      "alignItems": "center",
                                                                      "justifyContent": "center",
                                                                    }
                                                  } />
                              </View>
                              <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        ellipsizeMode="tail"
                                        style={
                                                  Object {
                                                            "color": "#111",
                                                            "fontSize": 24,
                                                            "paddingBottom": 3,
                                                            "paddingHorizontal": 10,
                                                          }
                                        }>
                                        Dobra
                              </Text>
                              <View
                                        backgroundColor="#2DB338"
                                        style={
                                                  Object {
                                                            "alignItems": "center",
                                                            "borderRadius": 30,
                                                            "height": 15,
                                                            "justifyContent": "center",
                                                            "overflow": "hidden",
                                                            "width": 15,
                                                          }
                                        }>
                                        <View
                                                  style={
                                                            Object {
                                                                      "alignItems": "center",
                                                                      "justifyContent": "center",
                                                                    }
                                                  } />
                              </View>
                    </View>
          </ElevatedView>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AirQualityIndex
    index={1} />,
}
`;
